src/aligners/nw.rs

```diff
@@ -20,6 +21,119 @@ const INF: Cost = Cost::MAX / 2;
 type NWLayers<const N: usize> = Layers<N, Vec<Cost>>;
 
 impl<const N: usize> NW<AffineCost<N>> {
+    fn track_path(&self, layers: &mut Vec<NWLayers<N>>, a: &Sequence, b: &Sequence) -> PATH {
+        let mut path: PATH = vec![(layers.len(), layers[0].m.len())];
+        let mut x = layers.len();
+        let mut y = layers[0].m.len();
+        let mut t: isize = -1;
+        let mut save = |x: &usize, y: &usize| path.push((*x, *y));
+
+        while x + y > 0 {
+            let mut found = false;
+            if t == -1 {
+                if x * y > 0 {
+                    //x*y >0 == (x > 0 && y > 0)
+                    if a[x - 1] == b[y - 1] && layers[x].m[y] == layers[x - 1].m[y - 1] {
+                        //match
+                        x -= 1;
+                        y -= 1;
+                        save(&x, &y);
+                        found = true;
+                    }
+                    if !found {
+                        if let Some(sub) = self.cm.sub {
+                            if layers[x].m[y] == layers[x - 1].m[y - 1] + sub {
+                                //mismatch
+                                x -= 1;
+                                y -= 1;
+                                save(&x, &y);
+                                found = true;
+                            }
+                        }
+                    }
+                }
+                if x > 0 && !found {
+                    if let Some(ins) = self.cm.ins {
+                        if layers[x].m[y] == layers[x - 1].m[y] + ins {
+                            //insertion
```

Comment:
This is actually a deletion I think. (Can you swap order so that ins goes before del?)

Reply:
nope; horizontal is deletion. This means you skip a character of `A` without matching it to one of `B`